### ISSUE TITLE
aws_iam_policy_attachment: Attached policy not found

### ISSUE THREAD

**Issue by pracucci** [terraform-providers/terraform-provider-aws#9918]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.12.7
+ provider.aws v2.25.0
```

### Affected Resource(s)

- `aws_iam_policy_attachment`

### Terraform Configuration Files

```
resource "aws_iam_policy" "test-policy" {
  name = "test-policy"
  policy = <<EOF
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid":      "Test",
            "Effect":   "Allow",
            "Action":   [ "s3:ListAllMyBuckets" ],
            "Resource": "*"
        }
    ]
}
EOF
}

resource "aws_iam_policy_attachment" "test-policy" {
  name       = "test"
  roles      = [aws_iam_role.test-role.name]
  policy_arn = aws_iam_policy.test-policy.arn
}
```

### Expected Behavior

The policy is attached on the AWS side and `terraform apply` succeed.

### Actual Behavior

The policy is attached on the AWS side but `terraform apply` fails with:

```
Error: [WARN] Error attaching policy with IAM Policy Attachment attach-codedeploy:
– Error: Attached policy not found
```

### Important Factoids

- The attach works with `terraform-aws-provider` `2.24.0` and breaks with `2.25.0`
- The role upon which we're attaching the policy already has **more than 10 policies attached**. The [limit of 10 policies per role](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-limits.html) is an AWS soft limit and we got it increased to 20 just asking to raise it through the AWS limit increase request public form.

**Comment by stevenwilliamson** [comment]
For reference im hitting this same issue. Confirmed that downgrading the aws provider to 0.24.0 resolves the issue and allows the terraform run to complete without error.

**Comment by 4unt3r** [comment]
aws provider 2.26 also causes the problem.. downgrade to 2.24 works..

**Comment by ryndaniels** [comment]
This should be fixed in v2.28.1 that was released yesterday - please let us know if you all are still running into issues with this!
